crates/oxc_linter/src/rules/no_compare_neg_zero.rs

```diff
@@ -0,0 +1,120 @@
+use oxc_ast::{
+    ast::{Expression, UnaryOperator},
+    AstKind, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-compare-neg-zero): Do not use the {0} operator to compare against -0.")]
+#[diagnostic(severity(warning))]
+struct NoCompareNegZeroDiagnostic(&'static str, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoCompareNegZero;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow comparing against -0
+    ///
+    /// ### Why is this bad?
+    /// The rule should warn against code that tries to compare against -0,
+    /// since that will not work as intended. That is, code like x === -0 will
+    /// pass for both +0 and -0. The author probably intended Object.is(x, -0).
+    ///
+    /// ### Example
+    /// ```javascript
+    /// if (x === -0) {}
+    /// ```
+    NoCompareNegZero,
+    correctness
+);
+
+impl Rule for NoCompareNegZero {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::BinaryExpression(expr) = node.get().kind() else { return; };
+        if let Some(op) =
+            OPERATORS_TO_CHECK.iter().find(|op| **op == expr.operator.as_str()).copied()
```

Comment:
Don't copy eslint on this one, we have all the tools we need ;-)

https://github.com/Boshen/oxc/blob/867f879483dfb7e3bcbda800db61cae55dcbbf35/crates/oxc_ast/src/ast/operator.rs#L184-L197

Reply:
Done.